From 5796c94b1eba4fb7407868555e630b409340b78e Mon Sep 17 00:00:00 2001
From: Aleksey <rhash.admin@gmail.com>
Date: Sat, 11 Sep 2021 21:23:34 +0300
Subject: [PATCH] fix incorrect total message in check-embedded mode

---
 hash_check.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hash_check.c b/hash_check.c
index 0c0ed84..0ed88ce 100644
--- a/hash_check.c
+++ b/hash_check.c
@@ -1147,7 +1147,7 @@ static int check_embedded_crc32(file_t* file)
 			log_error_file_t(&rhash_data.out_file);
 			res = -2;
 		} else if (!rhash_data.stop_flags) {
-			if (res >= 0)
+			if (res == 0)
 				rhash_data.ok++;
 			else if (res == -1 && errno == ENOENT)
 				rhash_data.miss++;
-- 
2.33.0

