diff -Naur RHash-1.4.2.org/configure RHash-1.4.2.sw/configure
--- RHash-1.4.2.org/configure	2022-02-25 19:11:47.060000000 +0000
+++ RHash-1.4.2.sw/configure	2022-02-25 19:19:40.480000000 +0000
@@ -402,6 +402,7 @@
           ia64) HOST_ARCH=ia64 ;;
           macppc|ppc*|Power*) HOST_ARCH=ppc ;;
           alpha) HOST_ARCH=alpha ;;
+          sw_64) HOST_ARCH=sw_64 ;;
           sun4*|sparc*) HOST_ARCH=sparc ;;
           parisc*|hppa*|9000*) HOST_ARCH=hppa ;;
           aarch64*) HOST_ARCH=aarch64 ;;
diff -Naur RHash-1.4.2.org/librhash/byte_order.h RHash-1.4.2.sw/librhash/byte_order.h
--- RHash-1.4.2.org/librhash/byte_order.h	2022-02-25 19:11:47.060000000 +0000
+++ RHash-1.4.2.sw/librhash/byte_order.h	2022-02-25 19:18:16.820000000 +0000
@@ -60,7 +60,7 @@
 /* try detecting endianness by CPU */
 #ifdef RHASH_BYTE_ORDER
 #elif defined(CPU_IA32) || defined(CPU_X64) || defined(__ia64) || defined(__ia64__) || \
-      defined(__alpha__) || defined(_M_ALPHA) || defined(vax) || defined(MIPSEL) || \
+      defined(__alpha__) || defined(_M_ALPHA) || defined(__sw_64__) || defined(_M_SW_64) || defined(vax) || defined(MIPSEL) || \
       defined(_ARM_) || defined(__arm__)
 #  define RHASH_BYTE_ORDER RHASH_BYTE_ORDER_LE
 #elif defined(__sparc) || defined(__sparc__) || defined(sparc) || \
diff -Naur RHash-1.4.2.org/librhash/test_hashes.h RHash-1.4.2.sw/librhash/test_hashes.h
--- RHash-1.4.2.org/librhash/test_hashes.h	2022-02-25 19:11:47.070000000 +0000
+++ RHash-1.4.2.sw/librhash/test_hashes.h	2022-02-25 19:19:15.490000000 +0000
@@ -89,6 +89,12 @@
 #ifdef __ia64__
 	" __ia64__"
 #endif
+#ifdef __sw_64__
+	" __sw_64__"
+#endif
+#ifdef _M_SW_64
+	" _M_SW_64"
+#endif
 #ifdef __alpha__
 	" __alpha__"
 #endif
